pkg/appmanager/outputConfig.go

```diff
@@ -61,26 +119,105 @@ func (appMgr *Manager) outputConfigLocked() {
-	for vKey, virtual := range resources.Virtuals {
-		for _, irule := range virtual.IRules {
-			if strings.Contains(irule, sslPassthroughIRuleName) {
-				clientProfCt := virtual.GetProfileCountByContext(customProfileClient)
-				serverProfCt := virtual.GetProfileCountByContext(customProfileServer)
-				if 0 == clientProfCt && 0 == serverProfCt {
-					sslProf := "Common/clientssl"
-					resources.Virtuals[vKey].AddFrontendSslProfileName(sslProf)
+	for partition, partitionConfig := range resources {
+		for vKey, virtual := range partitionConfig.Virtuals {
+			for _, irule := range virtual.IRules {
+				if strings.Contains(irule, sslPassthroughIRuleName) {
+					clientProfCt := virtual.GetProfileCountByContext(customProfileClient)
+					serverProfCt := virtual.GetProfileCountByContext(customProfileServer)
+					if 0 == clientProfCt && 0 == serverProfCt {
+						sslProf := "Common/clientssl"
+						resources[partition].Virtuals[vKey].AddFrontendSslProfileName(sslProf)
+					}
 				}
 			}
 		}
 	}
+
 	for _, profile := range appMgr.customProfiles.profs {
-		resources.CustomProfiles = append(resources.CustomProfiles, profile)
+		initPartitionData(resources, profile.Partition)
+		resources[profile.Partition].CustomProfiles = append(resources[profile.Partition].CustomProfiles, profile)
 	}
 	for _, irule := range appMgr.irulesMap {
-		resources.IRules = append(resources.IRules, *irule)
+		initPartitionData(resources, irule.Partition)
+		resources[irule.Partition].IRules = append(resources[irule.Partition].IRules, *irule)
 	}
 	for _, intDg := range appMgr.intDgMap {
-		resources.InternalDataGroups = append(resources.InternalDataGroups, *intDg)
+		initPartitionData(resources, intDg.Partition)
+		resources[intDg.Partition].InternalDataGroups = append(resources[intDg.Partition].InternalDataGroups, *intDg)
+	}
+
+	// Update resources to conform to the CCCL schema and empty out unneeded fields
+	// so they will be stripped out by the JSON marshaller
+	for partition, partitionConfig := range resources {
+		for i, _ := range partitionConfig.Virtuals {
+
+			resources[partition].Virtuals[i].Enabled = true
+
+			// Add the profiles to the Virtual Server
+			mode := strings.ToLower(resources[partition].Virtuals[i].Mode)
+			if mode == "http" {
+				resources[partition].Virtuals[i].IpProtocol = "tcp"
+				profile := ProfileRef{Partition: "Common", Name: "http", Context: "all"}
+				resources[partition].Virtuals[i].Profiles = append(resources[partition].Virtuals[i].Profiles, profile)
+			} else if mode == "tcp" {
+				resources[partition].Virtuals[i].IpProtocol = "tcp"
+				profile := ProfileRef{Partition: "Common", Name: "tcp", Context: "all"}
+				resources[partition].Virtuals[i].Profiles = append(resources[partition].Virtuals[i].Profiles, profile)
+			} else if mode == "udp" {
+				resources[partition].Virtuals[i].IpProtocol = "udp"
+			}
+
+			// Parse the SSL profile into partition and name
+			if nil != resources[partition].Virtuals[i].SslProfile {
```

Comment:
There is already a function that makes it easier to deal with Virtual.SslProfile. Calling resources[partition].Virtuals[i].GetFrontendSslProfileNames() will return an array that you can just loop through here.

Reply:
OK!